pilot/pkg/networking/util/util.go

```diff
@@ -359,16 +360,30 @@ func AddConfigInfoMetadata(metadata *core.Metadata, config config.Meta) *core.Me
 			FilterMetadata: map[string]*structpb.Struct{},
 		}
 	}
-	s := "/apis/" + config.GroupVersionKind.Group + "/" + config.GroupVersionKind.Version + "/namespaces/" + config.Namespace + "/" +
-		strcase.CamelCaseToKebabCase(config.GroupVersionKind.Kind) + "/" + config.Name
+
+	buf := bytesbufpool.GetWithCap(250)
+	defer bytesbufpool.Put(buf)
+
+	// format: /apis/{group}/{version}/namespaces/{namespace}/{kind}/{name}
+	buf.WriteString("/apis/")
+	buf.WriteString(config.GroupVersionKind.Group)
+	buf.WriteString("/")
+	buf.WriteString(config.GroupVersionKind.Version)
+	buf.WriteString("/namespaces/")
+	buf.WriteString(config.Namespace)
+	buf.WriteString("/")
+	buf.WriteString(strcase.CamelCaseToKebabCase(config.GroupVersionKind.Kind))
+	buf.WriteString("/")
+	buf.WriteString(config.Name)
```

Comment:
not sure how this inluence the benchmark, can you have a test

Reply:
Thanks, benchmark this change:
```
goos: darwin
goarch: amd64
pkg: istio.io/istio/pilot/pkg/networking/util
cpu: Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
                         │   old.txt   │              new.txt              │
                         │   sec/op    │   sec/op     vs base              │
AddConfigInfoMetadata-12   330.8n ± 0%   344.2n ± 2%  +4.05% (p=0.002 n=6)

                         │  old.txt   │              new.txt              │
                         │    B/op    │    B/op     vs base               │
AddConfigInfoMetadata-12   240.0 ± 0%   176.0 ± 0%  -26.67% (p=0.002 n=6)

                         │  old.txt   │              new.txt              │
                         │ allocs/op  │ allocs/op   vs base               │
AddConfigInfoMetadata-12   5.000 ± 0%   4.000 ± 0%  -20.00% (p=0.002 n=6)
```

Reply:
The usage seems pretty complicated. Did you see any issues with this apart from the micro benchmark?

Reply:
Only need to write once, the complexity seems acceptable. I noticed this because I collected it in production environment and it consumed about 2% of memory at some point in time.